pkg/kritis/util/allowlist.go

```diff
@@ -39,20 +41,93 @@ func RemoveGloballyAllowedImages(images []string) []string {
 	return notAllowlisted
 }
 
-func imageInAllowlist(image string) (bool, error) {
-	for _, w := range constants.GlobalImageAllowlist {
-		allowlistRef, err := name.ParseReference(w, name.WeakValidation)
+// RemoveGloballyAllowedImages returns all images that aren't in gap allowlists
+func RemoveGapAllowedImages(images []string, allowlist []string) []string {
+	notAllowlisted := []string{}
+	for _, image := range images {
+		allowlisted, err := imageInGapAllowlist(image, allowlist)
+		if err != nil {
+			glog.Errorf("couldn't check if %s is in gap allowlist: %v", image, err)
+		}
+		if !allowlisted {
+			notAllowlisted = append(notAllowlisted, image)
+		}
+	}
+	return notAllowlisted
+}
+
+// Do an image match based on reference.
```

Comment:
What's a reference?

Reply:
basically the way allowlist is handled in kritis is based on `name.ParseReference` and compare the references. For this newly added allowlist in GAP, i'm going to follow BinAuthz and use prefix matching. But maintain this reference-based matching for ISP.